inorbit_edge/robot.py

```diff
@@ -536,63 +558,145 @@ def _stop_cameras_streaming(self):
             for s in self.camera_streamers.values():
                 s.stop()
 
-    def publish_map(
-        self,
-        file,
-        map_id="map",
-        frame_id="map",
-        x=0,
-        y=0,
-        resolution=0.05,
-        ts=None,
-        is_update=False,
-    ):
-        """
-        Sends the information about a map and optionally its contents. The map message
-        data is passed as a separate variable to indicate that the map data needs to
-        be actually sent. An is_update flag can be provided to inform in the message
-        that this is an update message, aimed to correct a previous one, currently
-        used for frame_id/map_id updates.
-        """
+    def _read_map_file(self, file: str) -> Optional[bytes]:
```

Comment:
Why don't we make this a method of `MapData` ?  We can make `MapData` encapsulate all the details about getting the pixels data, calculating hashes, and computing sizes.

This way we can soon extend it to support not only files, but other sources, like in memory images.

Reply:
We can also have a `map_hash()` method that returns the hash. This way you can encapsulate this map related functions in that class and just pass around `MapData` objects instead of passing `MapData` and a separate sequence of bytes. That will also allow you to cache the hash instead of computing it for map request messages and map publishing

Reply:
That sounds good
I added a class for handling that. It caches pixels, hash and dimensions and determines validity of it based on the last modification time of the file.